ocaml/typing/typecore.ml

```diff
@@ -3459,8 +3499,8 @@ let remaining_function_type ty_ret mode_ret rev_args =
                newty2 ~level
                  (Tarrow (arrow_desc, ty_arg, ty_ret, commu_ok))
              in
-             let mode_ret =
-               Alloc.join (mode_fun :: closed_args)
+             let mode_ret, _ =
+               Alloc.newvar_above (Alloc.join (mode_fun :: closed_args))
```

Comment:
Why the `newvar_above` here?

Reply:
Because `mode_ret` is later used to construct `arrow_desc`, which requires `allowed` on both left and right.

Now, you might ask: why does `arrow_desc` require `Mode.Alloc.lr` for both argument and return? It should be right-only for argument, and left-only for return. And that's because we can't tell the variance of arrow types locally.